### PR TITLE
Addition of AxisymmetricRZ fill from input vector into RankFourTensor

### PR DESCRIPTION
**Pull request by sapitts**
As part of addressing Issue #5142 this commit is of additional lines to the RankFourTensor util in the tensor mechanics module to create an Axisymmetric C_{ijkl} tensor.  This new axisymmetric elasticity tensor will be used by the AxisymmetricRZ material to be migrated over to tensor mechanics later this week.

The axisymmetric elasticity tensor includes five independent quantities (ref: Slaughter, W.S. The Linearized Theory of Elasticity, pg. 209).  Following the procedure used for the symmetric and antisymmetric cases, the fillAxisymmetricRZFromInputVector reads in an input vector of size 5, creates a vector of size 9, and passes that new input vector to fillSymmetricFromInputVector.

### CODE REVIEW

**Comment by dschwen** [comment]
Can you squash the _delete.clang_complete_ commits into your first commit? Which editor/completion plugin are you using by the way?

File: modules/tensor_mechanics/include/utils/RankFourTensor.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ class RankFourTensor
     general_isotropic,
     symmetric_isotropic,
     antisymmetric_isotropic,
+    axisymmetricrz,
```
Comment (dschwen) [inline modules/tensor_mechanics/include/utils/RankFourTensor.h]: If you add an option here, you **must** change ```RankFourTensor::fillMethodEnum()``` accordingly.

File: modules/tensor_mechanics/src/utils/RankFourTensor.C [hunk 2]
```diff
@@ -741,6 +744,25 @@ RankFourTensor::fillSymmetricIsotropicFromInputVector(const std::vector<Real> &
 }
 
 void
+RankFourTensor::fillAxisymmetricRZFromInputVector(const std::vector<Real> & input)
+{
+    if (input.size() != 5)
```
Comment (dschwen) [inline modules/tensor_mechanics/src/utils/RankFourTensor.C]: We indent using 2 spaces

File: modules/tensor_mechanics/src/utils/RankFourTensor.C [hunk 3]
```diff
@@ -741,6 +744,25 @@ RankFourTensor::fillSymmetricIsotropicFromInputVector(const std::vector<Real> &
...
+        mooseError("To use fillAxisymmetricRZFromInputVector, your input must have size 5.  Your vector has size " << input.size());
+    std::vector<Real> input9;
```
Comment (dschwen) [inline modules/tensor_mechanics/src/utils/RankFourTensor.C]: I suggest to construct the vector with
```
std::vector<Real> input9(9);
```
and then directly set the components:
```
input9[0] = input[0];
...
```

**Comment by sapitts** [comment]
Thanks Daniel.  I'll make those changes to the spacing and google how to squash the clang deletions.  I'm using the Atom Editor you mentioned on the moose-users blog last week.

**Comment by dschwen** [comment]
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
look for _interactive rebase_ and _fixup_

**Comment by permcody** [comment]
Stephanie, you don't have to close your PRs if they fail. You can fix them up and force push them to update and trigger another round of testing without losing any of the discussion history.

**Pull request closed without merging**